src/Microsoft.AspNet.Mvc.TagHelpers/CacheTagHelper.cs

```diff
@@ -0,0 +1,266 @@
+        /// <summary>
+        /// Gets or sets the name of a HTTP request header to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByHeaderAttributeName)]
+        public string VaryByHeader { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of query parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByQueryAttributeName)]
+        public string VaryByQuery { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of route data parameters to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByRoute { get; set; }
+
+        /// <summary>
+        /// Gets or sets a comma-delimited set of cookie names to vary the cached result by.
+        /// </summary>
+        [HtmlAttributeName(VaryByRouteAttributeName)]
+        public string VaryByCookie { get; set; }
+
+        /// <summary>
+        /// Gets or sets a value that determines if the content is to be varied by the Identity for the logged in 
+        /// <see cref="HttpContext.User"/>.
+        /// </summary>
+        [HtmlAttributeName(VaryByUserAttributeName)]
+        public bool VaryByUser { get; set; }
+
+        /// <summary>
+        /// Gets or sets the exact <see cref="DateTimeOffset"/> the cache entry should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresOnAttributeName)]
+        public DateTimeOffset? ExpiresOn { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration from the time the cache entry when it should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresAfterAttributeName)]
+        public TimeSpan? ExpiresAfter { get; set; }
+
+        /// <summary>
+        /// Gets or sets the duration from last access that the cache entry should be evicted.
+        /// </summary>
+        [HtmlAttributeName(ExpiresSlidingAttributeName)]
+        public TimeSpan? ExpiresSliding { get; set; }
+
+        /// <summary>
+        /// Gets or sets the <see cref="CachePreservationPriority"/> policy for the cache entry.
+        /// </summary>
+        [HtmlAttributeName(CachePriorityAttributeName)]
+        public CachePreservationPriority? Priority { get; set; }
+
+        /// <inheritdoc />
+        public override async Task ProcessAsync(TagHelperContext context, TagHelperOutput output)
+        {
+            var key = GenerateKey(context);
+            string result;
+            if (!MemoryCache.TryGetValue(key, out result))
+            {
+                result = await context.GetChildContentAsync();
+                MemoryCache.Set(key, cacheSetContext =>
+                {
+                    UpdateCacheContext(cacheSetContext);
+                    return result;
+                });
+            }
+
+            // Clear the contents of the "cache" element since we don't want to render it.
```

Comment:
@NTaylorMullen none of the other MVC tag helpers have much in common w/ the `<cache/>` helper, so I'm not sure what guidance would carry over.  this is the first time we're dealing with a helper that does not target an existing HTML element and that always throws away its own begin and end tags.

put another way, nuking `TagName`, `PreContent`, `PostContent`, and completely replacing `Content` is _exactly_ what this helper should do.